### PR TITLE
[Fixes #8278] Tweak `@api private`.

### LINKED ISSUE #8278: Should we mark some modules as private before v1?
When reviewing #8274 I realized that it might be a good idea to mark some internal modules and classes as "@api private" before v1. Or is that too late or futile?

### PR DESCRIPTION
**Pull request by marcandre**
Many internal Modules and Classes marked as private.
Severity marked as public

### CODE REVIEW

**Comment by marcandre** [comment]
Unfinished though...

**Comment by marcandre** [comment]
I split the PR in two commits. The second one adds `@api private` to all cops and formatters. I though of writing a cop but was lazy and did it with search and replace. I hope I didn't miss too many things with the first one.

I don't know if anyone wants to review this? or maybe we should merge and someone can review the result?

File: lib/rubocop/cli/command/auto_genenerate_config.rb [hunk 1]
```diff
@@ -4,6 +4,9 @@ module RuboCop
   class CLI
     module Command
       # Generate a configuration file acting as a TODO list.
+      # @api private
```
Comment (bbatsov) [inline lib/rubocop/cli/command/auto_genenerate_config.rb:7]: I guess the duplicate is an oversight, right?

**Comment by bbatsov** [comment]
> The second one adds @api private to all cops and formatters.

Isn't this an overkill? It seems to me that such things are private by definition. But I guess it's still needed to generate the yard docs correctly.

File: lib/rubocop/cli/command/execute_runner.rb [hunk 2]
```diff
@@ -4,6 +4,9 @@ module RuboCop
   class CLI
     module Command
       # Run all the selected cops and report the result.
+      # @api private
+      #
+      # @api private
```
Comment (bbatsov) [inline lib/rubocop/cli/command/execute_runner.rb:9]: Same here.

File: lib/rubocop/cli/command/auto_genenerate_config.rb (hunk 1 above)
Comment (marcandre) [reply lib/rubocop/cli/command/auto_genenerate_config.rb:7]: Oops, search and replace error...

**Comment by marcandre** [comment]
> > The second one adds @api private to all cops and formatters.
> 
> Isn't this an overkill? It seems to me that such things are private by definition. But I guess it's still needed to generate the yard docs correctly.

I don't know what's best tbh. Note that we share constants from some Cops to other Cops (e.g. `AbcSizeCalculator` uses `CyclomaticComplexity::COUNTED_NODES`), so it's not that clear. Do I need to repeat that some of these constants are Sets, others Arrays, and that changing the type is a potential source of breakage unless something is done about that?

I'm completely neutral about the 2nd commit, and of course even the multiple `@api private` of the first commit won't necessarily change all that much, but shouldn't hurt much either...

Let me know how we should proceed.

**Comment by marcandre** [comment]
Added a small 3rd commit for some constants that can be made private. I'm just starting to realize that normal constants should probably be private by default...

**Comment by bbatsov** [comment]
Yeah, I was also thinking that constants should probably be private for real. It's common sense, but I've rarely seen Rubyists do this, as public by default has always been the mantra in our community. There are some merge conflicts that need to be addressed and you can merge this. I don't think any of the changes can hurt.

**Pull request merged**